Sources/Crashes/CrashReportSender.swift

```diff
@@ -18,29 +18,88 @@
 
 import Foundation
 import MetricKit
+import Common
 
-public final class CrashReportSender {
+public protocol CrashReportSending {
+    var pixelEvents: EventMapping<CrashReportSenderError>? { get }  // TODO: This should _not_ be optional
+
+    init(platform: CrashCollectionPlatform, pixelEvents: EventMapping<CrashReportSenderError>?)
+
+    func send(_ crashReportData: Data, crcid: String?) async -> (result: Result<Data?, Error>, response: HTTPURLResponse?)
+    func send(_ crashReportData: Data, crcid: String?, completion: @escaping (_ result: Result<Data?, Error>, _ response: HTTPURLResponse?) -> Void)
+}
+
+public enum CrashReportSenderError: Error {
+    case crcidMissing
+    case submissionFailed(HTTPURLResponse?)
+}
+
+// By conforming to a protocol, we can sub in mocks more easily
+public final class CrashReportSender: CrashReportSending {
+    static let reportServiceUrl = URL(string: "https://9e3c-20-75-144-152.ngrok-free.app/crash.js")!
+
+    static let httpHeaderCRCID = "crcid"
 
-    static let reportServiceUrl = URL(string: "https://duckduckgo.com/crash.js")!
     public let platform: CrashCollectionPlatform
+    public var pixelEvents: EventMapping<CrashReportSenderError>?
 
-    public init(platform: CrashCollectionPlatform) {
+    private let session = URLSession(configuration: .ephemeral)
+
+    public init(platform: CrashCollectionPlatform, pixelEvents: EventMapping<CrashReportSenderError>?) {
         self.platform = platform
+        self.pixelEvents = pixelEvents
     }
 
-    public func send(_ crashReportData: Data) async {
+    public func send(_ crashReportData: Data, crcid: String?, completion: @escaping (_ result: Result<Data?, Error>, _ response: HTTPURLResponse?) -> Void) {
         var request = URLRequest(url: Self.reportServiceUrl)
         request.setValue("text/plain", forHTTPHeaderField: "Content-Type")
         request.setValue(platform.userAgent, forHTTPHeaderField: "User-Agent")
+
+        request.setValue(crcid, forHTTPHeaderField: CrashReportSender.httpHeaderCRCID)
+        Logger.general.debug("Configured crash report HTTP request with crcid: \(crcid ?? "nil")")
+
         request.httpMethod = "POST"
         request.httpBody = crashReportData
 
-        do {
-            _ = try await session.data(for: request)
-        } catch {
-            assertionFailure("CrashReportSender: Failed to send the crash report")
+        Logger.general.debug("CrashReportSender: Awaiting session data")
+        let task = session.dataTask(with: request) { data, response, error in
+            if let response = response as? HTTPURLResponse {
+                Logger.general.debug("CrashReportSender: Received HTTP response code: \(response.statusCode)")
+                if response.statusCode == 200 {
+                    response.allHeaderFields.forEach { print("\($0.key): \($0.value)") }    // TODO: Why do we straight-up print these, rather than debug logging?
```

Comment:
@samsymons @loremattei @ayoy Does it make sense for me to convert this to a `Logger.general.debug` call instead of a simple `print`?

Reply:
@jdjackson Yep, definitely - that'll allow for more control over filtering etc. in the Xcode console.